### PR TITLE
feat(server): add topic naming strategy interface

### PR DESCRIPTION
**Pull request by william-reynolds-flosports**
Topic naming strategies allow the consumer to build strategies for
naming topics. Strategies may perform any actions necessary to construct
the necessary topic name, which is handy in circumventing the
limitations of nest event pattern decorators (arguments must be static
because the get json stringified).

### CODE REVIEW

File: lib/strategies/basic-topic-naming-strategy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import * as fc from 'fast-check';
+import { BasicTopicNamingStrategy } from './basic-topic-naming-strategy';
+
+describe(BasicTopicNamingStrategy.name, () => {
+    it('returns its argument unmodified', () => {
+        const strat = new BasicTopicNamingStrategy();
+
+        // prettier-ignore
+        fc.assert(fc.property(
```
Comment (haroldwaters) [inline lib/strategies/basic-topic-naming-strategy.spec.ts:9]: Ah man now I gotta go back and do the other strategies like this!

File: lib/interfaces.ts [hunk 2]
```diff
@@ -45,6 +46,31 @@ export interface SubscriptionNamingStrategy {
...
+ * behavior.
+ *
+ * @example
+ * class PrefixedTopicNameStrategy implements TopicNamingStrategy {
+ *     generateTopicName(originalName: string): string {
+ *         return `prefix-${originalName}`
+ *     }
+ * }
+ */
+export interface TopicNamingStrategy {
```
Comment (haroldwaters) [inline lib/interfaces.ts:70]: Mine eyes have seen the glory of the documentation of the lord.

**Review by haroldwaters — APPROVED** [review]

File: lib/strategies/basic-topic-naming-strategy.spec.ts (hunk 1 above)
Comment (william-reynolds-flosports) [reply lib/strategies/basic-topic-naming-strategy.spec.ts:9]: Be careful, it's addictive. 👁️

**Pull request merged**